.github/workflows/pr.yml

```diff
@@ -21,6 +21,15 @@ jobs:
         run: |
           pip install poetry
           poetry install --with dev
+      - name: Check code quality
+        if: ${{ always() }}
```

Comment:
Why do you need `if: ${{ always() }}` here and in the step below?

Reply:
But if the `Install Python dependencies` fails, your subsequent steps won't execute since `poetry` won't be available. Probably, you need to delete it from `Check code quality` but keep in other steps